### ISSUE TITLE
"DateTimeMock::__construct()"  side-effect using "idate()" when setting microseconds...

### ISSUE THREAD

**Issue by lfmimo** [slope-it/clock-mock#7]
Hi,

when using a non GMT **DateTimezone** and setting microseconds via

```
$this->setTime(
                idate('H', $this->getTimestamp()),
                idate('i', $this->getTimestamp()),
                idate('s', $this->getTimestamp()),
                (int) ClockMock::getFrozenDateTime()->format('u')
            );
```

the hour/minute is wrongly modified as **idate()** works on unix GMT-based timestamp.
Example:

Let's assume 02.04.2022 18:00:00 as Berlin time.
```
$dt = (new DateTime('now', new DateTimeZone('Europe/Berlin')))
```
DateTime now contains:

`date: 2022-04-02 18:00:00.0 Europe/Berlin (+02:00)`

Now using

```
$dt->setTime(
                idate('H', $dt->getTimestamp()),
                idate('i', $dt->getTimestamp()),
                idate('s', $dt->getTimestamp())
);
```

`$dt` contains:

`date: 2022-04-02 16:00:00.0 Europe/Berlin (+02:00)`.

This means the time is now set 2 hours back, but still the Europe/Berlin timezone.

**Solution in DateTimeMock:**

Use

```
$this->setTime(
                  (int) $this->format('H'),
                  (int) $this->format('i'),
                  (int) $this->format('s'),
                  (int) ClockMock::getFrozenDateTime()->format('u')
);
```

as this does't modify hour/minute/second when setting microseconds...

-mimo

**Comment by asprega** [comment]
Hi! Thanks for your report.

Would you be able to open a pull request with the fix and a test for it?

**Comment by lfmimo** [comment]
Sure ;)

**Comment by lfmimo** [comment]
Think I don't have sufficient access-rights to the repo:

<img width="1371" alt="image" src="https://user-images.githubusercontent.com/33455201/161508556-43d98ccc-b870-4d4a-915c-7eb01f06bb69.png">

**Comment by asprega** [comment]
You should first fork the repository and create your own copy, and then open the PR against the main repo, as described here: https://docs.github.com/en/get-started/quickstart/contributing-to-projects

hope this helps!

**Comment by lfmimo** [comment]
I've created the PR...

**Comment by asprega** [comment]
I just released version 0.2.2 including your contribution. Thanks 🎉
